type-tests/the-tests.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  dasherize,
+  camelize,
+  capitalize,
+  classify,
+  decamelize,
+  underscore,
+  w,
+} from '@ember/string';
+
+// @ts-expect-error
+dasherize();
+dasherize('blue man group'); // $ExpectType string
```

Comment:
This isn't doing anything! As noted in Discord, we need to:

```sh
yarn add -D expect-type
```

And then:

```suggestion
expectTypeOf(dasherize('blue man group')).toBeSTring();
```

And so on throughout.

Reply:
I just noticed there is a goofy `$ExpectType` syntax in a trailing comment -- maybe that was a macro or something?

Reply:
updated

Reply:
DefinitelyTyped uses [dtslint](https://github.com/Microsoft/DefinitelyTyped-tools/tree/master/packages/dtslint#write-tests) for type tests, which uses TSLint under the hood to do a form of type testing.